src/build_helper/src/fs/mod.rs

```diff
@@ -45,11 +51,34 @@ pub fn recursive_remove<P: AsRef<Path>>(path: P) -> io::Result<()> {
     #[cfg(not(windows))]
     let is_dir_like = fs::Metadata::is_dir;
 
-    if is_dir_like(&metadata) {
-        fs::remove_dir_all(path)
-    } else {
-        try_remove_op_set_perms(fs::remove_file, path, metadata)
+    const MAX_RETRIES: usize = 5;
+    const RETRY_DELAY_MS: u64 = 10;
+
+    let try_remove = || {
+        if is_dir_like(&metadata) {
+            fs::remove_dir_all(path)
+        } else {
+            try_remove_op_set_perms(fs::remove_file, path, metadata.clone())
+        }
+    };
+
+    // Retry deletion a few times to handle transient filesystem errors.
+    // This is unusual for local file operations, but it's a safeguard against
+    // edge cases like temporary locks (e.g., on Windows), slow antivirus scans,
```

Comment:
Nit: can you reword this to sth like

> ... but it's a mitigation against unlikely events where malware scanners may be holding a file beyond our control, to give the malware scanners some opportunity to release their hold

(Since it's really a mitigation and does not safeguard anything at all.)

Other cases like "temporary locks" or "race conditions where another process/thread is interacting with the file" are genuine bugs that are not the intended reason for introducing a retry here.